### DISCUSSION TITLE
Line Update

### DISCUSSION THREAD

**Discussion by RoyalFoxy** [ArthurSonzogni/FTXUI#282] [Q&A] [answered]
I'm currently searching the web for a way to "rewrite" or rather update a line I previously out out onto the terminal. If it's the same line it's simply `\r` in java at least. Now I don't only want to update the line I'm on... Rather multiple lines
Now FTXUI seems to be able to update multiple lines
Ive read about escape codes such as `\033[A` and `\033[F` but they seem to not work at all in java
How did you do it? If I could guess you either did you with the Cursus lib or with escape codes, if you used latter please tell me which ones you used it would be so helpful for me

**Comment by ArthurSonzogni** [comment, marked as answer]
Hello:

I am using:
https://github.com/ArthurSonzogni/FTXUI/blob/master/src/ftxui/screen/screen.cpp
```cpp
static const char MOVE_LEFT[] = "\r";
static const char MOVE_UP[] = "\x1B[1A";
static const char CLEAR_LINE[] = "\x1B[2K";

std::string Screen::ResetPosition(bool clear) {
  std::stringstream ss;
  if (clear) {
    ss << MOVE_LEFT << CLEAR_LINE;
    for (int y = 1; y < dimy_; ++y) {
      ss << MOVE_UP << CLEAR_LINE;
    }
  } else {
    ss << MOVE_LEFT;
    for (int y = 1; y < dimy_; ++y) {
      ss << MOVE_UP;
    }
  }
  return ss.str();
}
```

With \x1 being the byte with value 1.

More information on the specification:
https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h3-VT52-Mode
